src/index.ml

```diff
@@ -615,6 +623,7 @@ struct
           Mutex.unlock t.merge_lock;
           `Completed
       | `Aborted ->
+          t.pending_cancel <- false;
```

Comment:
I think this is not necessary, as the signalling `clear` does this after getting the `merge_lock`.
```suggestion
```